internal/command/registry/vulnsummary.go

```diff
@@ -0,0 +1,171 @@
+package registry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/samber/lo"
+	"github.com/spf13/cobra"
+
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/internal/render"
+	"github.com/superfly/flyctl/iostreams"
+)
+
+func newVulnSummary() *cobra.Command {
+	const (
+		usage = "vulnsummary <vulnid> ... [flags]"
+		short = "Show a summary of possible vulnerabilities in registry images"
+		long  = "Summarize possible vulnerabilities in registry images in an org, by app.\n" +
+			"Limit scanning to a single app if specified. Limit scanning to images\n" +
+			"used by running machines if specified. Limit reporting to\n" +
+			"specific vulnerability IDs or severities if specified."
+	)
+	cmd := command.New(usage, short, long, runVulnSummary,
+		command.RequireSession,
+		command.LoadAppNameIfPresent,
+	)
+
+	cmd.Args = cobra.ArbitraryArgs
+	flag.Add(
+		cmd,
+		flag.App(),
+		flag.Org(),
+		flag.Bool{
+			Name:        "running",
+			Shorthand:   "r",
+			Description: "Only scan images for running machines",
+		},
+		flag.String{
+			Name:        "severity",
+			Shorthand:   "S",
+			Description: fmt.Sprintf("Report only issues with a specific severity %v", allowedSeverities),
+		},
+	)
+
+	return cmd
+}
+
+func runVulnSummary(ctx context.Context) error {
+	var err error
+	filter, err := argsGetVulnFilter(ctx)
+	if err != nil {
+		return err
+	}
+
+	imgs, err := argsGetImages(ctx)
+	if err != nil {
+		return err
+	}
+
+	// fetch all image scans.
+	// TODO: spinner for long running fetches.
+	ios := iostreams.FromContext(ctx)
+	imageScan := map[string]*Scan{}
+	token := ""
+	tokenAppID := ""
+	for _, img := range imgs {
+		if _, ok := imageScan[img.Path]; ok {
+			continue
+		}
+
+		if img.AppID != tokenAppID {
+			tokenAppID = img.AppID
+			token, err = makeScantronToken(ctx, img.OrgID, img.AppID)
```

Comment:
Before I merge https://github.com/superfly/web/pull/3244, you could use `read_organization_apps` as the profile for a base token and then add the `AppFeatures` caveat to limit it further. That way you wouldn't have to fetch a separate token for each app. See [here](https://github.com/superfly/web/blob/master/app/models/tokens/profiles.rb) for the available profiles.